scm-webapp/src/main/java/sonia/scm/api/v2/resources/UserToUserDtoMapper.java

```diff
@@ -66,9 +66,7 @@ UserDto createDto(User user) {
     if (UserPermissions.modify(user).isPermitted()) {
       linksBuilder.single(link("update", resourceLinks.user().update(user.getName())));
       linksBuilder.single(link("publicKeys", resourceLinks.user().publicKeys(user.getName())));
-      if (userManager.isTypeDefault(user)) {
-        linksBuilder.single(link("password", resourceLinks.user().passwordChange(user.getName())));
-      }
+      linksBuilder.single(link("password", resourceLinks.user().passwordChange(user.getName())));
```

Comment:
Don't know if there is a better way. I need the password link for both internal and external user because on conversion of an external user a password must be set.

Reply:
We should really make sure that the api is consistent and correct. Because passwords cannot be changed/set for external users, the link for this action should not be returned in this case. It is okay if we have to perform an extra fetch if it means we can keep our api clean.